In [4]:
import numpy as np
import pandas as pd
import matplotlib as plt
import seaborn as sns
import matplotlib.pyplot as plt

## Q1)

In [5]:
data_train = pd.read_csv('../data/lemmatizer_tfidf.csv')
data_train.head()

id keyword location                                               text  \
0   1     NaN      NaN  our deeds are the reason of this may allah for...   
1   4     NaN      NaN             forest fire near la ronge sask  canada   
2   5     NaN      NaN  all residents asked to  shelter in place  are ...   
3   6     NaN      NaN  13 000 people receive evacuation orders in cal...   
4   7     NaN      NaN  just got sent this photo from ruby as smoke fr...   

   target           hashtags mentions urls  \
0       1         earthquake      NaN  NaN   
1       1                NaN      NaN  NaN   
2       1                NaN      NaN  NaN   
3       1          wildfires      NaN  NaN   
4       1  alaska, wildfires      NaN  NaN   

                                              tokens  \
0  ['deed', 'reason', 'may', 'allah', 'forgive', ...   
1  ['forest', 'fire', 'near', 'la', 'ronge', 'sas...   
2  ['resident', 'asked', 'shelter', 'place', 'not...   
3  ['13', '000', 'people', 'receive', 'evacuation...   
4  ['got', 'sent', 'photo', 'ruby', 'smoke', 'pou...   

                                              vector  
0                            [0. 0. 0. ... 0. 0. 0.]  
1                            [0. 0. 0. ... 0. 0. 0.]  
2                            [0. 0. 0. ... 0. 0. 0.]  
3  [0.         0.45036895 0.         ... 0.      ...  
4                            [0. 0. 0. ... 0. 0. 0.]

In [6]:
data_train.head(), data_train.shape
data_train['target'].value_counts()

0    4342
1    3271
Name: target, dtype: int64

In [7]:
df = data_train.drop(columns=['target'])
df.head()

id keyword location                                               text  \
0   1     NaN      NaN  our deeds are the reason of this may allah for...   
1   4     NaN      NaN             forest fire near la ronge sask  canada   
2   5     NaN      NaN  all residents asked to  shelter in place  are ...   
3   6     NaN      NaN  13 000 people receive evacuation orders in cal...   
4   7     NaN      NaN  just got sent this photo from ruby as smoke fr...   

            hashtags mentions urls  \
0         earthquake      NaN  NaN   
1                NaN      NaN  NaN   
2                NaN      NaN  NaN   
3          wildfires      NaN  NaN   
4  alaska, wildfires      NaN  NaN   

                                              tokens  \
0  ['deed', 'reason', 'may', 'allah', 'forgive', ...   
1  ['forest', 'fire', 'near', 'la', 'ronge', 'sas...   
2  ['resident', 'asked', 'shelter', 'place', 'not...   
3  ['13', '000', 'people', 'receive', 'evacuation...   
4  ['got', 'sent', 'photo', 'ruby', 'smoke', 'pou...   

                                              vector  
0                            [0. 0. 0. ... 0. 0. 0.]  
1                            [0. 0. 0. ... 0. 0. 0.]  
2                            [0. 0. 0. ... 0. 0. 0.]  
3  [0.         0.45036895 0.         ... 0.      ...  
4                            [0. 0. 0. ... 0. 0. 0.]

## Q2)

In [8]:
def get_missing_rate(df):
    print("Nombres de lignes dupliqués : ", df.duplicated().sum())
    print("Pourcentage des valeurs manquantes dans chaque colonnes:")
    print(df.isna().sum()/df.shape[0] * 100)

- Train :

In [9]:
get_missing_rate(df)

Nombres de lignes dupliqués :  0
Pourcentage des valeurs manquantes dans chaque colonnes:
id           0.000000
keyword      0.801261
location    33.272035
text         0.000000
hashtags    77.104952
mentions    73.610929
urls        47.839222
tokens       0.000000
vector       0.000000
dtype: float64


Notre dataset contient environ 33 % de valeurs manquantes dans la colonne location. Supprimer ces lignes ne serait pas pertinent, car cela reviendrait à perdre un tiers de nos données.

Étant donné que notre objectif est de classer des tweets, les variables les plus utiles sont keyword et text. La localisation n’a pas d’impact direct sur le contenu ou le sens du tweet, donc nous pouvons laisser cette colonne vide sans nuire à notre analyse.

Nous poursuivons donc l’exploration des données sans imputer ou supprimer les valeurs manquantes de location.